src/argon2.c

```diff
@@ -175,15 +175,15 @@ int argon2_hash(const uint32_t t_cost, const uint32_t m_cost,
 
     /* Detect and reject overflowing sizes */
     /* TODO: This should probably be fixed in the function signature */
```

Comment:
I guess this refers to the fact that function signature takes `size_t` (whose size is guaranteed >=16bits, but arch specific), but argon2_context and the rest of the code operates on `uint32_t`. There are casts sprinkled around in several places for this.
I think the TODO remark is still valid.